src/image/release.py

```diff
@@ -8,222 +8,91 @@
-        print(f"Exceptionally converting tag {base_tag} to {latest_alias}.")
-        release_tags[latest_alias] = revision
-        release_tags.pop(base_tag)
+    args = parser.parse_args()
 
-    # stable risks have an alias with any risk string
-    if base_tag.endswith("_stable"):
-        stable_alias = "_".join(base_tag.split("_")[:-1])
-        print(f"Adding stable tag alias {stable_alias} for {base_tag}")
-        release_tags[stable_alias] = revision
+    img_name = (
+        args.image_name
+        if args.image_name
+        else os.path.abspath(args.image_trigger).split("/")[-2]
+    )
 
-# we finally have all the OCI tags to be released,
-# and which revisions to release for each tag. Let's release!
-group_by_revision = defaultdict(list)
-for tag, revision in sorted(release_tags.items()):
-    group_by_revision[revision].append(tag)
+    print(f"Preparing to release revision tags for {img_name}")
+    all_revision_tags = shared.get_all_revision_tags(args.all_revision_tags)
+    revision_to_track = shared.get_revision_to_track(all_revision_tags)
+    print(
+        "Revision (aka 'canonical') tags grouped by revision:\n"
+        f"{json.dumps(revision_to_track, indent=2)}"
+    )
+    print(f"Reading all previous releases from {args.all_releases}...")
+    all_releases = shared.get_all_releases(args.all_releases)
+    tag_mapping_from_all_releases = shared.get_tag_mapping_from_all_releases(
+        all_releases
+    )
+    print(f"Parsing image trigger {args.image_trigger}")
+    image_trigger = shared.get_image_trigger(args.image_trigger)
+    tag_mapping_from_trigger, all_releases = shared.get_tag_mapping_from_trigger(
+        image_trigger, all_releases
+    )
+    print(
+        "Going to update channels according to the following:\n"
+        f"{json.dumps(tag_mapping_from_trigger, indent=2)}"
+    )
 
-print(
-    "Processed tag aliases and ready to release the following revisions:\n"
-    f"{json.dumps(group_by_revision, indent=2)}"
-)
-for revision, tags in group_by_revision.items():
-    revision_track = revision_to_track[revision]
-    source_img = (
-        "docker://ghcr.io/"
-        f"{args.ghcr_repo}/{img_name}:{revision_track}_{revision}"
+    # combine all tags
+    all_tags_mapping = {
+        **tag_mapping_from_all_releases,
+        **tag_mapping_from_trigger,
+    }
+    tag_to_revision = shared.get_tag_to_revision(
+        tag_mapping_from_trigger, all_tags_mapping, revision_to_track
+    )
+    release_tags = shared.get_releases_tags(tag_to_revision)
+    group_by_revision = shared.get_group_by_revision(release_tags)
+    print(
+        "Processed tag aliases and ready to release the following revisions:\n"
+        f"{json.dumps(group_by_revision, indent=2)}"
     )
     this_dir = os.path.dirname(__file__)
-    print(f"Releasing {source_img} with tags:\n{tags}")
-    subprocess.check_call(
-        [f"{this_dir}/tag_and_publish.sh", source_img, img_name] + tags
+    shared.upload_image(
```

Comment:
why is this function shared? What other script would be uploading images to the registries?